docs/analyzer/adding_recognizers.md

```diff
@@ -0,0 +1,247 @@
+
+![Recognizers class diagram](../assets/recognizers_class_diagram.png)
+
+- The `EntityRecognizer` is an abstract class for all recognizers.
+- The `RemoteRecognizer` is an abstract class for calling external PII detectors.
+See more info [here](#creating-a-remote-recognizer).
+- The `LocalRecognizer` is an abstract class for all recognizers running within the Presidio-analyzer process.
+- The `PatternRecognizer` is an class for supporting regex and deny-list based recognition logic,
+including validation (e.g., with checksum) and context support. See an example [here](#simple-example).
+
+## Extending the analyzer for additional PII entities
+
+First, a class based on `EntityRecognizer` needs to be created.
+Second, the new recognizer should be added to the recognizer registry,
+so that the `AnalyzerEngine` would be able to use the new recognizer during analysis.
+
+### Simple example
+
+For simple recognizers based on regular expressions or deny-lists,
+we can leverage the provided `PatternRecognizer`:
+
+```python
+from presidio_analyzer import PatternRecognizer
+titles_recognizer = PatternRecognizer(supported_entity="TITLE",
+                                      deny_list=["Mr.","Mrs.","Miss"])
+```
+
+Calling the recognizer itself:
+
+```python
+titles_recognizer.analyze(text="Mr. Schmidt",entities="TITLE")
+```
+
+Adding it to the list of recognizers:
+
+```python
+from presidio_analyzer import AnalyzerEngine, RecognizerRegistry
+
+registry = RecognizerRegistry()
+registry.load_predefined_recognizers()
+
+# Add the recognizer to the existing list of recognizers
+registry.add_recognizer(titles_recognizer)
+
+# Set up analyzer with our updated recognizer registry
+analyzer = AnalyzerEngine(registry=registry)
+
+# Run with input text
+text="His name is Mr. Jones"
+results = analyzer.analyze(text=text, language="en")
+print(results)
+
+```
+
+Alternatively, we can add the recognizer directly to the existing registry:
+
+```python
+from presidio_analyzer import AnalyzerEngine
+
+analyzer = AnalyzerEngine()
+
+analyzer.registry.add_recognizer(titles_recognizer)
+
+results = analyzer.analyze(text=text,language="en")
+print(results)
+```
+
+### Creating a new `EntityRecognizer` in code
+
+To create a new recognizer via code:
+
+1. Create a new Python class which implements [LocalRecognizer](../../presidio-analyzer/presidio_analyzer/local_recognizer.py).
+(`LocalRecognizer` implements the base [EntityRecognizer](../../presidio-analyzer/presidio_analyzer/entity_recognizer.py) class)
+
+    This class has the following functions:
+
+    i. load: load a model / resource to be used during recognition
+
+    ```python
+    def load(self)
+    ```
+
+    ii. analyze: The main function to be called for getting entities out of the new recognizer:
+
+    ```python
+    def analyze(self, text, entities, nlp_artifacts)
+    ```
+
+    Notes:
+    1. Each recognizer has access to different NLP assets such as tokens, lemmas, and more.
```

Comment:
maybe lemmatizations? for those who are not familiar with the term ;)

Reply:
Added link to wiki